doc/source/devel/physics.rst

```diff
@@ -0,0 +1,97 @@
+
+     - For this approach, it's important to understand how to access an instance of a particular kind of :cpp:class:`!Method` at an arbitrary point in the code (after all :cpp:class:`!Method` classes have been constructed).
+       One can use :cpp:expr:`enzo::problem()->method("<name>")` to return a pointer to the instance of the :cpp:class:`!Method` class for which :cpp:func:`Method::name()` returns ``"<name>"``.
+       If no such instance can be found, the expression returns a ``nullptr``.
+       You then need to cast that pointer to the appropriate :cpp:class:`!Method` subclass before you access the information.
+
+     - At the time of writing, this approach is commonly used to store information encoded within the :cpp:class:`!EnzoMethodGrackle` class.
+       To access such information, one could write
+
+       ..  code-block:: c++
+
+           const EnzoMethodGrackle *ptr = static_cast<const EnzoMethodGrackle*>
+             (enzo::problem()->method("grackle"));
+           if (ptr != nullptr) {
+             // maybe do stuff with ptr->try_get_chemistry() ...
+           }
+
+       In practice, some convenience functions have been written to help with these sorts of operations like :cpp:expr:`enzo::grackle_method()` or :cpp:expr:`enzo::grackle_chemistry()`
+
+     - In principle, one could do something analogous involving subclasses of :cpp:class:`!Initial`, but that could potentially introduce problems during a simulation restart.
+
+Storing information in a ``Physics`` class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It's often most useful to encode configuration-information within a :cpp:class:`!Physics` class when there isn't an obvious single :cpp:class:`!Method` class where it should be stored.
+
+A particular scenario where this is relevant is when separate (somewhat-interchangable) :cpp:class:`!Method` classes implement different algorithms to model the same set of physics.
+For example, consider the storage/access of the dual-energy formalism configuration.
+Since this is mostly relevant in the context of a hydro-solver it may make sense to store this information in the :cpp:class:`!Method` class that encapsulates a hydro-solver.
+However, because Enzo-E has :cpp:class:`!Method` classes that implement different hydro-solvers (that use the dual-energy formalism), we instead encode this information in a :cpp:class:`!Physics` class.
+
+There are also scenarios where some configuration information isn't really associated with any singular :cpp:class:`!Method`.
+For example the Equation-Of-State is important to a number of different methods.
+Another example includes the Gravitational Constant - this is important in self-gravity solvers and external-potential solvers, which are implemented in different :cpp:class:`!Method` classes.
+
+General Tips
+============
+
+The general advise is to implement a :cpp:class:`!Physics` class so that it is immutable (after construction the instance's state doesn't change).
```

Comment:
```suggestion
The general advice is to implement a :cpp:class:`!Physics` class so that it is immutable (after construction the instance's state doesn't change).
```